### ISSUE TITLE
[Enhancement]: Matière coefficienté (Pour régler un bug btw)

### ISSUE THREAD

**Issue by oriionn** [PapillonApp/papillon-v7#456]
### Description du problème

# Problème de matière coefficienté (Moyenne)
Nous avons en ce moment beaucoup de report de bug en rapport avec les moyennes, cela est dû à plusieurs facteurs, dont les moyennes coefficientés. Cependant, PRONOTE ne nous fournit pas les coefficients des matières.

### Description de l'amélioration

Je propose donc de rajouter dans les paramètres différentes façons de calculer les moyennes, il faudrait proposer le choix entre
- Un calcul de moyenne pondéré
- Un calcul de moyenne imitant l'algorithme de PRONOTE (calcul actuel)
- Un calcul de moyenne pondéré, se basant sur les coefficients du programme officiel de l'Education Nationale
- Un calcul de moyenne imitant l'algo de PRONOTE, se basant sur les coefficients du programme officiel de l'Education Nationale

### Première (Générale et Technologique)
Histoire Géographie : 3
Ens. Scientifique (général) : 3
Mathématiques (techno): 3
LVA : 3
LVB : 3
EMC : 1
Spécialité arrêté : 8

### Terminale (Générale et Technologique)
Histoire Géographie : 3
Ens. Scientifique (général) : 3
Mathématiques (techno): 3
LVA : 3
LVB : 3
EPS : 6
EMC : 1

### Source : [L'étudiant](https://www.letudiant.fr/bac/coefficients-bac-par-matiere-et-serie.html#coef-control-continu)

Pour cela, nous pourrions mettre en oeuvre un système de onboarding plus complet, avec la précision de la LVA et LVB, et du type de calcul de moyenne que l'utilisateur veut.

**Comment by oriionn** [comment]
Il faudrait aussi penser à rajouter les coefficients pour les BTS etc, mais je me suis concentré que sur le Bac pour le moment (peut-être, car je dois bientôt passer le Bac blanc de FR TT).

**Comment by Nicolasfr340** [comment]
Oui, et ce serait intéressant un bouton switch/menu déroulant dans le menu des notes, facile d'accès pour pouvoir voir la moyenne bac du contrôle continu avec les coefficients du bac (et avec possibilité d'importer les moyennes de l'année précédente pour les terminales) 

![image](https://github.com/user-attachments/assets/ddf48fd8-d822-4672-9750-713c688aa07b)

**Comment by Nicolasfr340** [comment]
### Message originalement posté par @JyhuKo sur #446
(_Originally posted by @JyhuKo in https://github.com/PapillonApp/Papillon/issues/446#issuecomment-2547127277_)
Je le poste ici suite à une demande d'un contributeur




"je sais que sur EcoleDirecte c'est calculé comme ca (Jai vérifié manuellement ):
### calcul moyenne générale ED

Étapes :
1. Multiplier la moyenne de chaque matière par son coefficient.
   Produit de la matière = Moyenne de la matière × Coefficient

2. Additionner tous les produits des matières.
   Somme des produits = ∑(Moyenne × Coefficient)

6. Diviser la somme des produits par la somme des coefficients.
   Moyenne générale = Somme des produits / Somme des coefficients

7. Arrondir au centième si nécessaire.

Exemple de calcul de moyenne générale :
| Matière      | Moyenne | Coefficient | Produit (Moyenne × Coef) |
|--------------|---------|-------------|--------------------------|
| Maths        | 11,3    | 1           | 11,3                    |
| Français     | 15,4    | 2           | 30,8                    |

Somme des produits : 11,3 + 30,8 = 42,1
Somme des coefficients : 1 + 2 = 3
Moyenne générale : 42,1 / 3 = 14,03


### calcul moyenne matière ED

Étapes :
1. Mettre toutes les notes sur la même base (20 ou 10 ou peut importe).
2. Multiplier chaque note par son coefficient (par défaut, le coefficient est 1 s'il n'est pas précisé).
3. Additionner les produits des notes.
4. Diviser la somme des produits par la somme des coefficients.
8. Mettre la moyenne sur 20 si nécessaire.
9. Arrondir au centième si nécessaire.

Exemple de calcul de la moyenne d'une matière :
Notes et coefficients :
- 9/10 (coef 1) = 18/20
- 6,5/10 (coef 1) = 13/20
- 13,5/20 (coef 2) = 13,5
- 9,5/10 (coef 1) = 19/20

Produits des notes :
- 18 × 1 = 18
- 13 × 1 = 13
- 13,5 × 2 = 27
- 19 × 1 = 19

Somme des produits : 18 + 13 + 27 + 19 = 77
Somme des coefficients : 1 + 1 + 2 + 1 = 5

Moyenne de la matière : 77 / 5 = 15,4

_j'espère que c'est compréhensible_"

**Comment by Nicolasfr340** [comment]
### Description de la fonctionnalité

Dans **l'onglet BAC** il y aurait une **estimation des points déjà récupéré sur le contrôle continu** selon ces notes avec le détail de ce que chaque matière te fais gagner avec le coefficient de la matière

### Contexte supplémentaire

_No response_

_Originally posted by @luky4005 in https://github.com/PapillonApp/Papillon/issues/548_
